tests/blackbox_test.go

```diff
@@ -311,3 +335,40 @@ func outer(t *testing.T, test types.Test, negativeTests bool) error {
 	}
 	return nil
 }
+
+//Identify and replace $uuid<num> with correct UUID
+//Variables with matching <num> should have identical UUIDs
+func replaceUUIDVars(t *testing.T, str string, UUIDmap map[int]string) string {
+	var finalStr string
+
+	tokenizedArr := strings.Split(str, " ")
+	for _, str := range tokenizedArr {
```

Comment:
It looks like this only finds the first `$uuid` occurrence in the token. Perhaps running the below logic on the entire input (instead of tokenizing it) repeatedly until it doesn't contain `$uuid` would work better?

Reply:
I think this should handle all $uuid occurrences since the case of $uuid<num>$uuid<num> should never show up, right?

Reply:
We could have a config with no whitespace in it, since whitespace isn't needed in JSON. As a contrived example, this is a valid config:

```
{"ignition":{"version":"2.1.0"},"storage":{"filesystems":[{"mount":{"device":"/dev/sda","wipeFilesystem":false,"format":"btrfs","label":"data","uuid":"8A7A6E26-5E8F-4CCA-A654-46215D4696AC"}},{"mount":{"device":"/dev/sdb","wipeFilesystem":false,"format":"ext4","label":"more-data","uuid":"3e343b34-da88-4bfd-b8c6-673d950d082d"}}]}}
```

If I'm reading your code right, I think it wouldn't find the `$uuid` for the second uuid in there.

Reply:
Could use regexp to find all occurances which might be cleaner